pkg/raft/testdata/snapshot_succeed_via_app_resp_behind.txt

```diff
@@ -165,6 +165,6 @@ stabilize 1
 # This is despite its PendingSnapshot having been 12.
 status 1
 ----
-1: StateReplicate match=12 next=13
-2: StateReplicate match=12 next=13
-3: StateReplicate match=11 next=13 inflight=1
+1: StateReplicate match=12 next=13 sentCommit=0 matchCommit=11
```

Comment:
Looks like we should have an invariant `sentCommit >= matchCommit`, similarly to `Next > Match`.

Otherwise, the logic in `ShouldSendMsgApp` can send an unnecessary message in some cases. For example in this `sentCommit=0 matchCommit=11` state we will send a message even if `commit == 11`?